integration/integration_test.go

```diff
@@ -810,10 +858,24 @@ func TestSigsnoop(t *testing.T) {
 	t.Parallel()
 
 	sigsnoopCmd := &command{
-		name:           "StartSigsnoopGadget",
-		cmd:            fmt.Sprintf("$KUBECTL_GADGET trace signal -n %s", ns),
-		expectedRegexp: fmt.Sprintf(`%s\s+test-pod\s+test-pod\s+\d+\s+sh\s+SIGTERM`, ns),
-		startAndStop:   true,
+		name:         "StartSigsnoopGadget",
+		cmd:          fmt.Sprintf("$KUBECTL_GADGET trace signal -n %s -o json", ns),
+		startAndStop: true,
+		expectedOutputFn: func(output string) error {
+			expectedEntry := &signalTypes.Event{
+				Event:  buildBaseEvent(ns),
+				Comm:   "sh",
+				Signal: "SIGTERM",
+			}
+
+			return expectOnly(output, expectedEntry, func(e *signalTypes.Event) {
```

Comment:
The CI is sometimes failing because, in some cases, the gadget captures the event:

```json
{"node":"fv-az39-947","namespace":"test-sigsnoop-2744583553906244966","pod":"test-pod","container":"test-pod","type":"normal","pid":16075,"tpid":16075,"signal":"SIGURG","comm":"runc:[2:INIT]","mountnsid":4026532724}
```

From my understanding, this signal happens at the container creation, and sometimes the gadget takes more time to start capturing events than others, so it sometimes loses this event. I added a delay of 5 seconds after launching the gadget, and it still failed 7/30 tests I ran locally.

I am wondering what is the best approach here. Should I implement a helper `expectAtLeastOne` for this test? WDYT?

Reply:
I think having a `expectAtLeastOne` helps in the case events are generated outside our control.